src/Features/Core/Portable/Shared/TestHooks/Legacy/ListenerForwarders.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Runtime.CompilerServices;
+
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.AsynchronousOperationListener))]
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.AsynchronousOperationListenerProvider))]
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.EmptyAsyncToken))]
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.IAsynchronousOperationListenerProvider))]
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.IAsynchronousOperationWaiter))]
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.IAsynchronousOperationListener))]
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.IAsyncToken))]
+[assembly: TypeForwardedTo(typeof(Microsoft.CodeAnalysis.Shared.TestHooks.TaskExtensions))]
```

Comment:
Out of curiosity, why are these needed?  for F#/TS/JS?

Reply:
➡️ Considering the impact of other changes coming in this PR, I'm guessing these are not going to be necessary in the final revision. However, I decided to make the relocation separate from the removal of the original types.

Reply:
This change was intended to avoid the need to have **WorkspacesTestUtilities** reference `Features`. This is no longer possible, and **WorkspacesTestUtilities** ended up taking a reference all the way to **RemoteWorkspaces**. However, this change may still be beneficial as part of work to ensure all asynchronous operations started by Roslyn are tracked.